### PR TITLE
Backport compliance check from dav1d 1.4.0

### PR DESCRIPTION
**Pull request by fbossen**
Check for trailing marker/zero bits in tile data

### CODE REVIEW

**Review by kkysen — CHANGES_REQUESTED** [review]

File: src/msac.rs [hunk 1]
```diff
@@ -195,6 +195,16 @@ impl MsacContext {
...
 
+    pub unsafe fn last_bytes(&self, offset: isize) -> &[u8] {
+        let buf_pos = self.buf_pos.offset(offset);
+        let buf = unsafe {
+            let len = self.buf_end.offset_from(buf_pos) as usize;
+            std::slice::from_raw_parts(buf_pos, len)
+        };
+
+        buf
+    }
```
[the remark this answers is not in the document]
Comment (fbossen) [reply src/msac.rs:206]: Note that `offset` is generally negative here. The slice returned by `buf()` isn't large enough.

[the remark this answers is not in the document]
Comment (rinon) [reply src/msac.rs:206]: @kkysen what is the status here? looks like you abandoned #1074, can we just use this then?

[the remark this answers is not in the document]
Comment (kkysen) [reply src/msac.rs:206]: Sorry, didn't abandon that one. Just kept forgetting to fix it up.  I'll do that now.

[the remark this answers is not in the document]
Comment (kkysen) [reply src/msac.rs:206]: You can fix this PR now based on #1074, I think.  You should be able to use `.data()` and `.buf_index()` with `.wrapping_add_signed` and it'll all be safe code.

**Review by kkysen — APPROVED** [review]

**Pull request merged**